### PR TITLE
[e2e tests] Fix blocks tests failing with Gutenberg active

### PR DESCRIPTION
**Pull request by adimoldovan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).

### Changes proposed in this Pull Request:

Fixes blocks e2e tests that are failing when Gutenberg is active and the editor canvas is within an iframe.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure the e2e tests that are using the editor are passing with and without Gutenberg. 
2. Check CI, all tests should pass. See test runs: [with Gutenberg](https://github.com/woocommerce/woocommerce/actions/runs/8805465182/job/24168838038?pr=46842), without Gutenberg.

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/global-setup.js [hunk 1]
```diff
@@ -57,9 +56,6 @@ module.exports = async ( config ) => {
 	const adminPage = await adminContext.newPage();
 	const customerPage = await customerContext.newPage();
 
-	// Ensure live mode state (coming soon = no)
-	await setOption( request, baseURL, 'woocommerce_coming_soon', 'no' );
```
Comment (adimoldovan) [inline plugins/woocommerce/tests/e2e-pw/global-setup.js:61]: This is already set in the `test-env-setup.sh` script, so there is no need for it here. More than that, it fails when re-running tests locally and it is already set to 'no'.

File: .github/workflows/smoke-test-daily.yml [hunk 2]
```diff
@@ -406,7 +406,7 @@ jobs:
                     - plugin: 'Gutenberg'
                       slug: gutenberg
                     - plugin: 'Gutenberg - Nightly'
-                      slug: gutenberg
```
Comment (adimoldovan) [inline .github/workflows/smoke-test-daily.yml:409]: The slug here is used to create the report, and it was overwriting the stable gutenberg one.

File: plugins/woocommerce/package.json [hunk 3]
```diff
@@ -28,8 +28,8 @@
 		"env:restart": "pnpm wp-env destroy && pnpm wp-env start --update",
 		"env:start": "pnpm wp-env start",
 		"env:stop": "pnpm wp-env stop",
-		"env:test": "WP_ENV_LIFECYCLE_SCRIPT_AFTER_START='./tests/e2e-pw/bin/test-env-setup.sh' && pnpm env:dev && pnpm playwright install chromium",
```
Comment (adimoldovan) [inline plugins/woocommerce/package.json:31]: The lifecycle script is already defined in `.wp-env.json`, making this redundant.

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/create-cart-block.spec.js [hunk 4]
```diff
@@ -10,40 +15,11 @@ test.describe( 'Transform Classic Cart To Cart Block', () => {
 	test.use( { storageState: process.env.ADMINSTATE } );
 
 	test( 'can transform classic cart to cart block', async ( { page } ) => {
-		// go to create a new page
-		await page.goto( 'wp-admin/post-new.php?post_type=page' );
+		await goToPageEditor( { page } );
```
Comment (lanej0) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/create-cart-block.spec.js:18]: I'm not usually a fan of writing a helper to replace `page.goto`, I think it really reduces the readability of tests -- but it looks like we're also dismissing the welcome modal with this one... so I'm okay with it.

**Review by lanej0 — APPROVED** [review]

**Pull request merged**